genestack_client/genestack_connection.py

```diff
@@ -129,6 +129,21 @@ def login(self, email, password):
         if not logged['authenticated']:
             raise GenestackAuthenticationException("Fail to login %s" % email)
 
+    def login_by_token(self, token):
+        """
+        Attempt a login on the connection with the specified token.
+        Raises an exception if the login fails.
+
+        :param token: email
+        :rtype: None
+        :raises: :py:class:`~genestack_client.GenestackServerException` if module version is outdated
+                 :py:class:`~genestack_client.GenestackAuthenticationException` if login failed
+        """
+        self.check_version()
+        logged = self.application('genestack/signin').invoke('authenticateByToken', token)
+        if not logged['authenticated']:
+            raise GenestackAuthenticationException('Fail to login with token')
```

Comment:
Maybe dump token itself in that error message?
`'Fail to login with token "%s"'`

Reply:
No, this is insecure.